public/search/sql_async_search_interceptor.ts

```diff
@@ -77,18 +87,29 @@ export class SQLAsyncSearchInterceptor extends SearchInterceptor {
     };
 
     const onPollingSuccess = (pollingResult: any) => {
-      if (pollingResult && pollingResult.body.meta.status === 'SUCCESS') {
-        return false;
-      }
-      if (pollingResult && pollingResult.body.meta.status === 'FAILED') {
-        const jsError = new Error(pollingResult.data.error.response);
-        this.deps.toasts.addError(jsError, {
-          title: i18n.translate('queryEnhancements.sqlQueryError', {
-            defaultMessage: 'Could not complete the SQL async query',
-          }),
-          toastMessage: pollingResult.data.error.response,
-        });
-        return false;
+      if (pollingResult) {
+        const statusStr: string = (pollingResult.body.meta.status as string).toUpperCase();
+        const status = SparkJobState[statusStr as keyof typeof SparkJobState];
+        switch (status) {
+          case SparkJobState.SUCCESS:
+            return false;
+          case SparkJobState.FAILED:
+            const jsError = new Error(pollingResult.data.error.response);
+            this.deps.toasts.addError(jsError, {
+              title: i18n.translate('queryEnhancements.sqlQueryError', {
+                defaultMessage: 'Could not complete the SQL async query',
+              }),
+              toastMessage: pollingResult.data.error.response,
+            });
+            return false;
+          default:
+            if (request.params?.progress_query_id) {
```

Comment:
Apologies. I might be confused here and might need a little bit more context. 

Would you mind adding your approach details here: https://github.com/kavilla/queryEnhancements/issues/25? With how you intend this will be used in the rest of the application.

Once the request is here it is already in context of itself so if another request is fired I think it would have a different query ID and status. So here we are interceptor and won't return back a value to whoever invoked search until we want to return the dataframe value.

So it could maybe create the ID after this in here: https://github.com/kavilla/queryEnhancements/blob/main/public/search/sql_async_search_interceptor.ts#L114 and pass it to these functions. because it will be in context.

Reply:
The thing is that if we manage the request ID ourselves, we end up not able to communicate request process to the outside -- the caller needs to know what request to listen on ahead of time, since we don't return any intermediary results via the existing search interface.

Reply:
Newest version unconditionally sends status with a generated ID, resolving